doc/api/process.markdown

```diff
@@ -167,6 +167,55 @@ this, you can either attach a dummy `.catch(function() { })` handler to
 `resource.loaded`, preventing the `'unhandledRejection'` event from being
 emitted, or you can use the [`'rejectionHandled'`][] event.
 
+## Event: 'warn'
+
+Emitted whenever Node.js emits a "Process Warning". Process Warnings are
+similar to Errors in that they describe exceptional conditions that are being
+brought to the user's attention, they are not, however, part of the normal
+Node.js and JavaScript Error handling flow. Node.js currently emits Warnings
+whenever it detects: (a) the use of deprecated APIs, (b) known security
+vulnerabilities that are the result of poor or outdated programming practice,
+and (c) bad coding practices that could lead to sub-optimal application
+performance or bugs.
+
+The event handler for `'warn'` events is called with a single `warn` argument
+whose value is a `Warning` object. `Warning` objects have the following
+properties:
+
+* `name` - The name of the warning (currently one of `DeprecationWarning`,
+  `SecurityWarning`, or `BadPracticeWarning`).
+* `message` - A system-provided description of the warning.
+* `stack` - A stack trace to the location in the code where the warning was
```

Comment:
We might want to describe how to interact with these parameters, namely `stack`. Not a big deal.